### PR TITLE
Feature/experimental sidebar features v2

### PR DESCRIPTION
**Pull request by reflog**

### CODE REVIEW

File: v4/source/channels.yaml [hunk 1]
```diff
@@ -2281,3 +2281,316 @@
           $ref: "#/components/responses/Unauthorized"
         "403":
           $ref: "#/components/responses/Forbidden"
+  "/users/{user_id}/teams/{team_id}/channels/categories":
+    get:
+      tags:
+        - channels
+      summary: Get user's sidebar categories
+      description: >
```
Comment (hmhealey) [inline v4/source/channels.yaml:2289]:
```suggestion
      description: >
      Get a list of sidebar categories that will appear in the user's sidebar
      on the given team, including a list of channel IDs in each category.
      
      __Minimum server version__: 5.26
      
```

File: v4/source/channels.yaml [hunk 2]
```diff
@@ -2281,3 +2281,316 @@
...
+          $ref: "#/components/responses/Unauthorized"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        "404":
+          $ref: "#/components/responses/NotFound"
+    post:
+      tags:
+        - channels
+      summary: Create user's sidebar category
+      description: >
```
Comment (hmhealey) [inline v4/source/channels.yaml:2327]:
```suggestion
      description: >
      Create a custom sidebar category for the user on the given team.
      
      __Minimum server version__: 5.26

```

File: v4/source/channels.yaml [hunk 3]
```diff
@@ -2281,3 +2281,316 @@
...
+          $ref: "#/components/responses/Unauthorized"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        "404":
+          $ref: "#/components/responses/NotFound"
+    put:
+      tags:
+        - channels
+      summary: Update user's sidebar categories
+      description: >
```
Comment (hmhealey) [inline v4/source/channels.yaml:2363]:
```suggestion
      description: >
      Update any number of sidebar categories for the user on the given team. This
      can be used to reorder the channels in these categories.
      
      __Minimum server version__: 5.26
      
```

File: v4/source/channels.yaml [hunk 4]
```diff
@@ -2281,3 +2281,316 @@
...
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        "404":
+          $ref: "#/components/responses/NotFound"
+  "/users/{user_id}/teams/{team_id}/channels/categories/order":
+    get:
+      tags:
+        - channels
+      summary: Get user's sidebar category order
+      description: >
```
Comment (hmhealey) [inline v4/source/channels.yaml:2400]:
```suggestion
      description: >
      Returns the order of the sidebar categories for a user on the given team as
      an array of IDs.
      
      __Minimum server version__: 5.26

```

File: v4/source/channels.yaml [hunk 5]
```diff
@@ -2281,3 +2281,316 @@
...
+          $ref: "#/components/responses/Unauthorized"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        "404":
+          $ref: "#/components/responses/NotFound"
+    put:
+      tags:
+        - channels
+      summary: Update user's sidebar category order
+      description: >
```
Comment (hmhealey) [inline v4/source/channels.yaml:2438]:
```suggestion
      description: >
      Updates the order of the sidebar categories for a user on the given team.
      The provided array must include the IDs of all categories on the team.
      
      __Minimum server version__: 5.26

```

File: v4/source/channels.yaml [hunk 6]
```diff
@@ -2281,3 +2281,316 @@
...
+            type: string
+        - name: user_id
+          in: path
+          description: User GUID
+          required: true
+          schema:
+            type: string
+      responses:
+        "200":
+          description: Category retrieval successful
```
Comment (hmhealey) [inline v4/source/channels.yaml:2419]:
```suggestion
          description: Order retrieval successful
```

File: v4/source/channels.yaml [hunk 7]
```diff
@@ -2281,3 +2281,316 @@
...
+          required: true
+          schema:
+            type: string
+        - name: user_id
+          in: path
+          description: User GUID
+          required: true
+          schema:
+            type: string
+      responses:
```
Comment (hmhealey) [inline v4/source/channels.yaml:2344]:
```suggestion
      requestBody:
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/SidebarCategory"
          required: true
      responses:
```

File: v4/source/channels.yaml [hunk 8]
```diff
@@ -2281,3 +2281,316 @@
...
```
Comment (hmhealey) [inline v4/source/channels.yaml:2380]:
```suggestion
      requestBody:
          content:
            application/json:
              schema:
                type: array
                items: 
                  $ref: "#/components/schemas/SidebarCategory"
          required: true
      responses:
```

File: v4/source/channels.yaml [hunk 9]
```diff
@@ -2281,3 +2281,316 @@
...
+        "200":
+          description: Category retrieval successful
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/OrderedSidebarCategories"
```
Comment (hmhealey) [inline v4/source/channels.yaml:2425]:
```suggestion
                  type: string
```

File: v4/source/channels.yaml [hunk 10]
```diff
@@ -2281,3 +2281,316 @@
...
+          required: true
+          schema:
+            type: string
+        - name: user_id
+          in: path
+          description: User GUID
+          required: true
+          schema:
+            type: string
+      responses:
```
Comment (hmhealey) [inline v4/source/channels.yaml:2455]:
```suggestion
      requestBody:
          content:
            application/json:
              schema:
                type: array
                items:
                  type: string
          required: true
      responses:
```

File: v4/source/channels.yaml [hunk 11]
```diff
@@ -2281,3 +2281,316 @@
...
+        "200":
+          description: Category creation successful
```
Comment (hmhealey) [inline v4/source/channels.yaml:2457]:
```suggestion
          description: Order update successful
```

File: v4/source/channels.yaml [hunk 12]
```diff
@@ -2281,3 +2281,316 @@
...
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/SidebarCategory"
```
Comment (hmhealey) [inline v4/source/channels.yaml:2461]:
```suggestion
                type: array
                items:
                  type: string
```

File: v4/source/channels.yaml [hunk 13]
```diff
@@ -2281,3 +2281,316 @@
...
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        "404":
+          $ref: "#/components/responses/NotFound"
+  "/users/{user_id}/teams/{team_id}/channels/categories/{category_id}":
+    get:
+      tags:
+        - channels
+      summary: Get sidebar category
+      description: >
```
Comment (hmhealey) [inline v4/source/channels.yaml:2475]:
```suggestion
      description: >
      Returns a single sidebar category for the user on the given team.
      
      __Minimum server version__: 5.26

```

File: v4/source/channels.yaml [hunk 14]
```diff
@@ -2281,3 +2281,316 @@
...
+          $ref: "#/components/responses/Unauthorized"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        "404":
+          $ref: "#/components/responses/NotFound"
+    put:
+      tags:
+        - channels
+      summary: Update sidebar category
+      description: >
```
Comment (hmhealey) [inline v4/source/channels.yaml:2517]:
```suggestion
      description: >
      Updates a single sidebar category for the user on the given team.
      
      __Minimum server version__: 5.26

```

File: v4/source/channels.yaml [hunk 15]
```diff
@@ -2281,3 +2281,316 @@
...
+          required: true
+          schema:
+            type: string
+        - name: category_id
+          in: path
+          description: Category GUID
+          required: true
+          schema:
+            type: string
+      responses:
```
Comment (hmhealey) [inline v4/source/channels.yaml:2540]:
```suggestion
      requestBody:
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/SidebarCategory"
          required: true
      responses:
```

File: v4/source/channels.yaml [hunk 16]
```diff
@@ -2281,3 +2281,316 @@
...
+          $ref: "#/components/responses/Unauthorized"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        "404":
+          $ref: "#/components/responses/NotFound"
+    delete:
+      tags:
+        - channels
+      summary: Delete sidebar category
+      description: >
```
Comment (hmhealey) [inline v4/source/channels.yaml:2559]:
```suggestion
      description: >
      Deletes a single sidebar category for the user on the given team. Only
      custom categories can be deleted.
      
      __Minimum server version__: 5.26

```

File: v4/source/definitions.yaml [hunk 17]
```diff
@@ -2703,6 +2703,62 @@ components:
...
+          type: string
+        display_name: 
+          type: string
+        type: 
+          type: string
+          enum:
+            - C
+            - U
+            - D
+            - F
```
Comment (hmhealey) [inline v4/source/definitions.yaml:2726]:
```suggestion
            - channels
            - custom
            - direct_messages
            - favorites
```

File: v4/source/definitions.yaml [hunk 18]
```diff
@@ -2703,6 +2703,62 @@ components:
...
```
Comment (hmhealey) [inline v4/source/definitions.yaml:2745]:
```suggestion
            - channels
            - custom
            - direct_messages
            - favorites
```

**Comment by reflog** [comment]
awesome. thanks @hmhealey

**Pull request merged**